frontend/src/notifications.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from "react";
+import { Button } from "@chakra-ui/react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBellSlash } from "@fortawesome/free-solid-svg-icons";
+
+type KeyType = number | string;
+
+type NotificationOptions<T> = {
+  filter: (x: T) => boolean;
+  message: (xs: T[]) => string;
+  key: (x: T) => KeyType;
+};
+
+export function useNotification<T>(
+  xs: T[],
+  { filter, message, key }: NotificationOptions<T>
+) {
+  const [seen, setSeen] = React.useState<KeyType[]>([]);
+  const unseen = xs.filter((x) => !seen.includes(key(x)));
+  const filtered = unseen.filter(filter);
+  React.useEffect(() => {
+    if (filtered.length > 0) {
+      const msg = message(filtered);
+      if (Notification.permission === "granted") {
+        new Notification(msg);
+        setSeen((seen) => [...seen, ...filtered.map(key)]);
```

Comment:
This means it will only ever notify about a given pull once per page load, even if that pull goes back and forth between needing review and being reviewed. Am I reading this correctly? If so, is that what we want?

Reply:
Oh, good point. I don't think we want that; I've got an idea.

Reply:
Okay, fixed in [2f6557c](https://github.com/iFixit/pulldasher/pull/327/commits/2f6557c1b7fa8469195220cf072502391594c232) and [5cfee99](https://github.com/iFixit/pulldasher/pull/327/commits/5cfee998ee7071ebd9e58bfad986469a4892cd06). Now we always update `seen` to be the list of PRs we _could_ have notified on last time it rendered. That means that if a PR becomes ineligible to be notified about it'll get dropped from the list, which means it'll get notified for again after it becomes eligible again.